### PR TITLE
Weekly view

### LINKED ISSUE #28: Week View
Not an issue but more of a friendly tip. 

My fork has a week view. It's not perfect but it works. I am able to switch to week and month view. If anyone would want this as an option let me know, I'll try to submit it as a PR.

alt text

https://github.com/jedt/react-native-calendar/blob/master/CalendarIOS.js#L231-L273

### PR DESCRIPTION
**Pull request by noitcudni**
image

Added a new property 'calendarFormat,' which can either be "monthly" or "weekly".

Simplified the eventMap data structure. Instead of having a map of maps, it's now just a map keyed by date (YYYY-MM-DD) string. 

The more interesting change is in getStartMoment(). 

Please let me know what you think and if you'd like me to clarify and/or clean up anything. 

Closes #28

### CODE REVIEW

**Comment by christopherdro** [comment]
Awesome! Thanks for the PR. Will try and review this weekend.

**Comment by christopherdro** [comment]
@noitcudni I am seeing a few issues with how the surrounding logic is works in this mode.

When I attempt to implement your changes on the example app of mater I see two issues that needs addressing before moving further.

- Dates are rendered incorrectly. 
- Month in header changes to the next month when it shouldn't.

demo2

File: components/Calendar.js [hunk 1]
```diff
@@ -156,63 +155,81 @@ export default class Calendar extends Component {
     }
   }
 
-  renderMonthView(argMoment, eventsMap) {
+  getStartMoment(calFormat, currMoment) {
+    const weekStart = this.props.weekStart;
+
+    let r;
```
Comment (christopherdro) [inline components/Calendar.js]: What is `r` signify here? Can we make this variable name more explicit?

File: components/Calendar.js [hunk 2]
```diff
@@ -94,44 +96,41 @@ export default class Calendar extends Component {
...
   selectDate(date) {
-    this.setState({ selectedMoment: date });
-    this.props.onDateSelect && this.props.onDateSelect(date ? date.format(): null );
+    this.setState({ selectedMoment: date,
+                    currentMonthMoment: date});
   }
 
   onPrev = () => {
-    const newMoment = moment(this.state.currentMonthMoment).subtract(1, 'month');
+    const newMoment = this.props.calendarFormat == 'monthly' ?
```
Comment (christopherdro) [inline components/Calendar.js]: strict equality `===`

File: components/Calendar.js [hunk 3]
```diff
@@ -94,44 +96,41 @@ export default class Calendar extends Component {
...
+            moment(this.state.currentMonthMoment).subtract(1, 'month'):
+            moment(this.state.currentMonthMoment).subtract(1, 'week');
     this.setState({ currentMonthMoment: newMoment });
     this.props.onTouchPrev && this.props.onTouchPrev(newMoment);
   }
 
   onNext = () => {
-    const newMoment = moment(this.state.currentMonthMoment).add(1, 'month');
+    const newMoment = this.props.calendarFormat == 'monthly' ?
```
Comment (christopherdro) [inline components/Calendar.js]: strict equality

File: components/Calendar.js [hunk 4]
```diff
@@ -156,63 +155,81 @@ export default class Calendar extends Component {
     }
   }
 
-  renderMonthView(argMoment, eventsMap) {
+  getStartMoment(calFormat, currMoment) {
+    const weekStart = this.props.weekStart;
+
+    let r;
+    if (calFormat == "monthly") {
```
Comment (christopherdro) [inline components/Calendar.js]: single quotes + strict equality

**Comment by noitcudni** [comment]
Thanks for the feedback. I'll take a look. I didn't know the option of swiping left and right was available.

File: components/Calendar.js (hunk 1 above)
Comment (noitcudni) [reply components/Calendar.js]: This is "borrowed" from the C convention. "r" is the return value, but I'll comply to you request.

**Comment by noitcudni** [comment]
@christopherdro 
Thanks again for you detailed screen shot (how did you make that by the way?)

The following is what I fixed.
1) Make `scrollEnded()` aware of the weekly mode.
2) Preinstantiate prior and next 2 weeks in `getMonthStack()`, which I renamed to `getStack()`.

**Comment by christopherdro** [comment]
@noitcudni Thanks for updating the PR. I will give it another go in the next days.

The app to create gif screenshots is called Licecap.
http://www.cockos.com/licecap/

**Comment by noitcudni** [comment]
I noticed that the calendar can actually be "scrolled" vertically, so I changed `automaticallyAdjustContentInsets={false}` inside the `ScrollView.` Swiping left and right still work, while scrolling vertically doesn't do anything any longer.

**Comment by amol-c** [comment]
Would love to see this PR get in.

**Comment by coderdave** [comment]
@noitcudni Can this be resolved with master?

**Comment by noitcudni** [comment]
I'll do it in the next couple days.

**Comment by coderdave** [comment]
Also, `this.state.currentMonthMoment` is undefined on line 285 in `Calendar.js`. Should it be `this.state.currentMoment` now?

**Comment by christopherdro** [comment]
Looks like the PR renames those state variables to exclude the word moment. I think we can leave that out for now as it's likely to continuously create conflicts with other PR's that might get merged before this one.

**Comment by noitcudni** [comment]
So, should I stay put and wait?

**Comment by GU5TAF** [comment]
Would love to see this get merged, has there been any progress @noitcudni @christopherdro ?

**Comment by noitcudni** [comment]
I have been busy at work. I might be able to start working on this again soon. @christopherdro, would you want me to get started on this soon? Or, do you still want me to hold off and wait for other pull requests to get pulled in first.

**Comment by christopherdro** [comment]
@GU5TAF @noitcudni Sorry for the late response. Been quite busy myself the last couple weeks.

I will spend some time this weekend looking over everything.

**Comment by christopherdro** [comment]
Would be nice to wait for this to get merged since it has a significant amount of changes.
https://github.com/christopherdro/react-native-calendar/pull/107

**Comment by joshjhargreaves** [comment]
What needs to be done to get this merged :)?

**Comment by noitcudni** [comment]
I might have some free time now to look into the merge conflicts. Are you guys still trying to prioritize other pull requests?

**Comment by noitcudni** [comment]
@coderdave @christopherdro I just did a 3 way merge from your master into my fork's master.  Please review at your earliest convenience.

**Review by christopherdro — APPROVED** [review]

**Comment by MarsWang42** [comment]
Hey @noitcudni thanks for the work! But it seems like it's not working properly on ios if I set `scrollenabled`. I currently on RN 0.44.0.

**Comment by noitcudni** [comment]
@Marswang92. What do you mean not working properly? Does it scroll but displays the wrong dates? Or, it doesn't scroll at all?

**Comment by MarsWang42** [comment]
So if I set `scrollEnabled`, the height of calendar will be set as the same as `monthly` and dates will not show up on the first render.
b7zumxncsz
`scrollEnabled` removed, everything works well.
ksaqibzsqs
@noitcudni Any clue?

**Comment by MarsWang42** [comment]
Hey, the rendering problem has been solved via setting `removeClippedSubviews` to be false, my bad. But still no clue about the height issue.

**Comment by MarsWang42** [comment]
@noitcudni found out the reason and made a PR https://github.com/christopherdro/react-native-calendar/pull/139/

Would you mind check?

**Comment by noitcudni** [comment]
@Marswang92 sure thing.

**Comment by jaycee425** [comment]
should the documentation be updated for these changes?

**Pull request merged**